webapp/Connector/src/app/store/StudyProducts.js

```diff
@@ -144,7 +169,13 @@ Ext.define('Connector.app.store.StudyProducts', {
                     });
                 }
                 product.other_products = otherProducts;
-                products.push(product);
+
+                if (this.validProduct(product.product_name)) {
+                    products.push(product);
+                }
+                else{
+                    this.secretData[product.product_id] = product;
```

Comment:
We need to exclude "None" products from being accessible on product detail page. Also, like mentioned before, I prefer to push every product into products, but have an extra flag to indicate if it should be shown.